### PR TITLE
Extend readme with environment based configuration

### PR DESCRIPTION
**Pull request by sldblog**
Environment based configuration is possible in `Hutch` but this was only apparent if someone took a look into `lib/hutch/config.rb`.

### Changes
This changeset updates the readme to mention the possibility of this and also provides a cheat sheet for the configuration preference order.

Easier to view changes: https://github.com/sldblog/hutch/blob/dl-update-configuration-readme/README.md#environment-variables

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -317,6 +317,20 @@ Known configuration parameters are:
...
+ * `connection_timeout` &rarr; `HUTCH_CONNECTION_TIMEOUT`.
+
+### Configuration precedence
+
+In order from lowest to highest precedence:
+
+0. Default values
+0. `HUTCH_*` environment variables
+0. Configuration file
+0. Explicit settings through `Hutch::Config.set`
```
Comment (sldblog) [inline README.md]: Could someone please confirm that the above order is correct?

Comment (michaelklishin) [inline README.md]: @olleolleolle can but it looks OK to me.

**Comment by michaelklishin** [comment]
Thank you!

**Comment by sldblog** [comment]
🎩 Thanks :)

**Pull request merged**